src/services/organizeImports.ts

```diff
@@ -322,96 +323,109 @@ function coalesceImportsWorker(importGroup: readonly ImportDeclaration[], compar
         return importGroup;
     }
 
-    const { importWithoutClause, typeOnlyImports, regularImports } = getCategorizedImports(importGroup);
-    const coalescedImports: ImportDeclaration[] = [];
-
-    if (importWithoutClause) {
-        coalescedImports.push(importWithoutClause);
-    }
-
-    for (const group of [regularImports, typeOnlyImports]) {
-        const isTypeOnly = group === typeOnlyImports;
-        const { defaultImports, namespaceImports, namedImports } = group;
-        // Normally, we don't combine default and namespace imports, but it would be silly to
-        // produce two import declarations in this special case.
-        if (!isTypeOnly && defaultImports.length === 1 && namespaceImports.length === 1 && namedImports.length === 0) {
-            // Add the namespace import to the existing default ImportDeclaration.
-            const defaultImport = defaultImports[0];
-            coalescedImports.push(
-                updateImportDeclarationAndClause(defaultImport, defaultImport.importClause.name, namespaceImports[0].importClause.namedBindings),
-            );
-
-            continue;
+    const importGroupsByAttributes = groupBy(importGroup, decl => {
+        if (decl.attributes) {
+            let attrs = decl.attributes.token + " ";
+            for (const x of sort(decl.attributes.elements)) {

```

Comment:
I’m not sure this `sort` is doing what you want. `elements` is an array of `Node` objects, which are going to get coerced to strings before sorting. Most likely every element is going to turn into `[object Object]` or `[object Node]` and the sort order will never change.

<img width="407" alt="image" src="https://github.com/microsoft/TypeScript/assets/3277153/990482de-43ef-4fe6-9106-05ef2e6ad5c9">